### PR TITLE
fix #10322: displaying hammer-on / pull-off while importing gp files

### LINKED ISSUE #10322: [MU4 Task] [GP] Display hammer-on/pull-off
**Task description**
1. Display hammer-on/pull-off with H and P (or H.O and P.O.)  indication

**Expected:**
Screenshot 2022-01-19 at 17 58 42

2. Continuous hammer-ons/pull-offs should have one tie over the whole passage

**Expected:**
Screenshot 2022-01-25 at 18 18 42

### PR DESCRIPTION
**Pull request by alexpavlov96**
Resolves: *(direct link to the issue)*

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: src/importexport/guitarpro/internal/gtp/gpconverter.cpp [hunk 1]
```diff
@@ -937,6 +937,20 @@ void GPConverter::addContiniousSlideHammerOn()
                 legatoSlides.erase(startNote);
                 legatoSlides[endNote] = slur;
             }
+
+            // TODO-gp: implement for editing too. Now works just for import.
+            if (slide.second == SlideHummerOn::HammerOn) {
```
Comment (Jojo-Schmitz) [inline src/importexport/guitarpro/internal/gtp/gpconverter.cpp:942]: `SlideH**u**mmerOn::H**a**mmerOn`?

**Review by RomanPudashkin — APPROVED** [review]

File: src/importexport/guitarpro/internal/gtp/gpconverter.cpp (hunk 1 above)
Comment (alexpavlov96) [reply src/importexport/guitarpro/internal/gtp/gpconverter.cpp:942]: haha right! once you read the same way, does it really matter? =)

Comment (Jojo-Schmitz) [reply src/importexport/guitarpro/internal/gtp/gpconverter.cpp:942]: Coukld you please get that "Hummer" changed to "Hammer" all the way?

File: src/importexport/guitarpro/internal/gtp/gpconverter.cpp [hunk 2]
```diff
@@ -937,6 +937,20 @@ void GPConverter::addContiniousSlideHammerOn()
...
+                Measure* measure = startNote->chord()->measure();
+
+                auto midTick = (startTick + endTick) / 2;
+                Segment* segment = measure->getSegment(SegmentType::ChordRest, midTick);
+                StaffText* staffText = Factory::createStaffText(segment);
+                QString hummerText = (startNote->pitch() > endNote->pitch()) ? "P" : "H";
+
+                staffText->setPlainText(hummerText);
```
Comment (Jojo-Schmitz) [inline src/importexport/guitarpro/internal/gtp/gpconverter.cpp:950]: and here of course too

Comment (Jojo-Schmitz) [reply src/importexport/guitarpro/internal/gtp/gpconverter.cpp]: Thanks !

**Pull request merged**